src/components/ExecuteButton.js

```diff
@@ -0,0 +1,183 @@
+           Satellite not registered for Playbook execution
+        </Text>
+        <Button
+            key="register"
+            // eslint-disable-next-line no-console
+            variant='link' onClick={ () => console.log('TODO: add link') }>
+            Learn how to register Satelite
+        </Button>
+    </TextContent>),
+    no_receptor: (<TextContent>
+        <Text component={ TextVariants.p }>
+            <ExclamationCircleIcon
+                className="ins-c-remediations-connection-status-error ins-c-remediations-connection-status"
+                aria-label="connection status" />
+            Cannot remediate - Receptor not configured
+        </Text>
+        <Text component={ TextVariants.small }>
+            Configure Receptor to automatically remediate
+        </Text>
+        <Button
+            key="configure"
+            // eslint-disable-next-line no-console
+            variant='link' onClick={ () => console.log('TODO: add link') }>
+            Learn how to configure
+        </Button>
+    </TextContent>)
+})[status];
+
+const ExecuteButton = ({ isLoading, data, getConnectionStatus, remediationId, issueCount }) => {
+    const [ open, setOpen ] = useState(false);
+    const [ isUserEntitled, setIsUserEntitled ] = useState(false);
+    useEffect(() => {
+        window.insights.chrome.auth.getUser().then(user => setIsUserEntitled(user.entitlements.smart_management.is_entitled));
+    }, []);
+
+    const [ connected, disconnected ] = data.reduce(
+        ([ pass, fail ], e) => (e.connection_status === 'connected' ? [ [ ...pass, e ], fail ] : [ pass, [ ...fail, e ] ])
+        , [ [], [] ]
+    );
+
+    const rows = [ ...connected, ...disconnected ].map(con =>
+        ({ cells: [ con.executor_name || 'Direct connection', con.system_count, { title: styledConnectionStatus(con.connection_status) }]})
+    );
+    const connectedCount = connected.reduce((acc, e) => e.system_count + acc, 0);
+    const systemCount = data.reduce((acc, e) => e.system_count + acc, 0);
+
+    const pluralize = (number, str) => number > 1 ? `${number} ${str}s` : `${number} ${str}`;
+    return (isUserEntitled
+        ?  <React.Fragment>
+            <Button
+                onClick={ () => { setOpen(true); getConnectionStatus(remediationId); } }>
+        Execute playbook
+            </Button>
+            <Modal
+                className="ins-c-dialog"
+                width={ '50%' }
+                title={ 'Execute playbook' }
+                isOpen={ open }
+                onClose={ () => setOpen(false) }
+                isFooterLeftAligned
+                actions={ [
+                    <Button
+                        key="confirm"
+                        variant="primary"
+                        disabled={ connected.length > 0 }
+                        onClick={ () => setOpen(false) }>
+                        { `Execute Playbook on ${pluralize(connectedCount, 'system')}` }
```

Comment:
according to the mockups, `Playbook` should be lower case

Reply:
The mockups are wrong. We use Playbook everywhere because it is sort of Ansible's trademark thing.

Reply:
In that case, do we want to capitalize `playbook` in the modal header?

Reply:
Yes, good catch!

Reply:
Going to update my review